app/services/activitypub/fetch_remote_poll_service.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class ActivityPub::FetchRemotePollService < BaseService
+  include JsonLdHelper
+
+  def call(poll, on_behalf_of = nil)
+    @json = fetch_resource(poll.status.uri, true, on_behalf_of)
+
+    return unless supported_context? && expected_type?
+
+    expires_at = begin
+      if @json['closed'].is_a?(String)
+        @json['closed']
+      elsif @json['closed'].is_a?(TrueClass)
+        Time.now.utc
+      else
+        @json['endTime']
+      end
+    end
+
+    items = begin
+      if @json['anyOf'].is_a?(Array)
+        @json['anyOf']
+      else
+        @json['oneOf']
+      end
+    end
+
+    poll.update!(
+      expires_at: expires_at,
+      cached_tallies: items.map { |item| item.dig('replies', 'totalItems') || 0 }
+    )
```

Comment:
Same remark as earlier, what about fetching collections that aren't inlined?
Furthermore, if the poll *has* changed since it was initially fetched, this code will produce inconsistent tallies.

Reply:
>Furthermore, if the poll has changed since it was initially fetched, this code will produce inconsistent tallies.

I don't understand what you mean. This is fetched from the origin, for remote polls. The origin has the most correct count of votes.

Reply:
Sorry, I meant that if the list of options changed (which nothing prevents), the updated tallies are going to be garbage.

Reply:
I don't think updating the options of published polls should be allowed. That invalidates the results.

Reply:
You could reorder them or add requested options, I don't know. I think we should at least update tallies based on `name`, not on the option's position in the serialization.